scoap3/cli_fixes.py

```diff
@@ -23,8 +27,58 @@ def init_default_location():
         loc = Location()
         loc.name = 'default'
         loc.default = True
-        loc.uri = '/virtualenv/files/'
+        loc.uri = os.environ.get('SCOAP_DEFAULT_LOCATION') or '/virtualenv/files/'
         db.session.add(loc)
         db.session.commit()
     else:
         error("Default location already exists.")
+
+
+@fixdb.command()
+@click.option(
+    '--dry-run',
+    is_flag=True,
+    default=False,
+    help='If set to True no changes will be committed to the database.')
+@click.argument(
+    'json_file',
+    type=click.Path(exists=True))
+@with_appcontext
+def fix_doi_dates(json_file, dry_run):
+    """
+    Fixes the imprint/publication/copyright dates on a list of DOIs.
+    """
+    records = RecordMetadata.query.all()
+    record_dois_uuids = {
+        rec.json['dois'][0]['value']: str(rec.id)
+        for rec in records
+    }
+
+    info('Records extracted.')
+
+    # dois to be changes, with the new dates
+    with open(json_file) as _file:
+        dois_with_dates = json.load(_file)
+
+    try:
+        for doi in dois_with_dates.keys():
+            uuid = record_dois_uuids[doi]
+            rec = Record.get_record(uuid)
+
+            date = dois_with_dates[doi]
+            year = int(date.split('-')[0])
+
+            rec['imprints'][0]['date'] = date
+            rec['publication_info'][0]['year'] = year
+            rec['copyright'][0]['year'] = year
```

Comment:
It would be good to log the exact changes made for each article even with `dry_run` specified, e.g. log for each article something like `[DOI]: [OLD_DATE] -> [NEW_DATE]` (or even add record id or uuid)